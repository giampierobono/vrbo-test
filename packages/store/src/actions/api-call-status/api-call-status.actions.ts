import { Action } from "redux";

export enum ApiCallStatusActions {
  PropertiesPollLoadingStart = "[ApiCallStatus] Properties call loading start",
  PropertiesPollLoadingEnd = "[ApiCallStatus] Properties call loading end"
}

export const propertiesPollLoadingStartAction = (): Action => ({
  type: ApiCallStatusActions.PropertiesPollLoadingStart
});

export const propertiesPollLoadingEndAction = (): Action => ({
  type: ApiCallStatusActions.PropertiesPollLoadingEnd
});
