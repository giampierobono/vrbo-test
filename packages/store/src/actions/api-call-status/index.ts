export * from "./api-call-status.actions";
