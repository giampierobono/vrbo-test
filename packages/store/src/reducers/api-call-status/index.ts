export * from "./api-call-status.reducer";
