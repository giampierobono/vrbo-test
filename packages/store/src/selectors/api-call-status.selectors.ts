import { GlobalState } from "../global-state.model";

export const getPropertiesListLoadingStatus = (state: GlobalState): boolean =>
  state.apiCallStatus.propertiesPollLoading;
