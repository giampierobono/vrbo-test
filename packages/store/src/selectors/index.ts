export * from "./api-call-status.selectors";
export * from "./property.selectors";
