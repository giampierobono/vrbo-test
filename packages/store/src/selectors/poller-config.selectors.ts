import { PollerConfig } from "@vrbo/data-models";
import { GlobalState } from "../global-state.model";

export const getPollerConfig = (state: GlobalState): PollerConfig =>
  state.pollerConfig.config;
