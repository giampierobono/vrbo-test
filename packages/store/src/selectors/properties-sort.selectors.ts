import { PropertiesSortByConfig } from "@vrbo/data-models";
import { GlobalState } from "../global-state.model";

export const getPropertiesSortByConfig = (
  state: GlobalState
): PropertiesSortByConfig => state.propertiesListSortBy.sortConfig;
