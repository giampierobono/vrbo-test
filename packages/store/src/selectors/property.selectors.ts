import { PropertiesSortBy, Property } from "@vrbo/data-models";
import { createSelector } from "reselect";
import { GlobalState } from "../global-state.model";

const getPropertiesList = (
  state: GlobalState,
  props: { sortBy: PropertiesSortBy }
) => state.propertiesList.properties;

export const makeGetOrderedPropertiesList = () =>
  createSelector(
    [getPropertiesList],
    (
      propertiesList: Property[],
      props: { sortBy: PropertiesSortBy }
    ): Property[] => {
      switch (props.sortBy) {
        case PropertiesSortBy.Title: {
          return propertiesList.sort((a, b) =>
            a.title === b.title ? 0 : a.title > b.title ? 1 : 0
          );
        }
        default: {
          return propertiesList;
        }
      }
    }
  );
