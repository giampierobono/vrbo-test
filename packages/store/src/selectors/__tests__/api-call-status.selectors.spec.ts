import { mockState } from "../../../../../jest-global-mocks";
import { getPropertiesListLoadingStatus } from "../api-call-status.selectors";

describe("Api call status selectors", () => {
  describe("getPropertiesListLoadingStatus", () => {
    it("should return correct value", () => {
      expect(getPropertiesListLoadingStatus(mockState)).toMatchSnapshot();
    });
  });
});
