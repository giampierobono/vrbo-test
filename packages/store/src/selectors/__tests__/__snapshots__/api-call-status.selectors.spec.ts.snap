// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api call status selectors getPropertiesListLoadingStatus should return correct value 1`] = `false`;
